### ISSUE TITLE
suggestion: `deno task update` and automatic updates

### ISSUE THREAD

**Issue by iuioiua** [denoland/saaskit#20]
An `update` task like [this](https://github.com/Hexagon/bundlee/blob/4858ed39e8220a41779c126b80ec918d43287979/deno.json#L9) would be fantastic. This'd also be great to integrate into a weekly GitHub action like [this](https://github.com/hayd/deno-udd/blob/master/.github/workflows/udd.yml) to keep dependencies up-to-date.

**Comment by niklasmtj** [comment]
I would have to dig a bit to understand how this works, but that action looks like a good idea. Until GitHub's dependabot don't understand Deno natively this should be a good idea to implement. 

I would like to grab this in the next couple days. Will be at KubeCon next week. So if someone else wants to grab this, go for it :)

**Comment by iuioiua** [comment]
Actually, it'd be best to have decent test coverage before proceeding with this feature. Otherwise, bugs caused by updated dependencies could fall through the cracks.

**Comment by iuioiua** [comment]
This is dependent on #267.

**Comment by iuioiua** [comment]
We're ready to pursue this. All Fresh dependencies (those that come along with Fresh) should only be updated manually.

**Comment by niklasmtj** [comment]
I started working on it. One thing that we have to discuss is the case of the [semantic versioning](https://github.com/hayd/deno-udd#semantic-versioning) and what kind of versions we would like to see. Because right now without any version flags for `udd` I will get `preact` in version 11, which is, as of this writing, in experimental mode. I don't think that we would like to directly update to those versions.

However, do we like to see those changes anyway or maybe also have two task definitions, one checking for "production" dependency updates and one for "experimental/major bumps"?

**Comment by iuioiua** [comment]
Yes, the 2nd option looks better. It's pretty much what I was talking about in my previous comment - better said, version pinning.

If pinning versions causes issues for cases like `preact`, then that's a dealbreaker. Our options would be to use a different dependency updater or contribute to `udd`, which, frankly, doesn't appear to be actively maintained. WDYT?

**Comment by niklasmtj** [comment]
Hmm, not being able to start the app with pinned versions really is a dealbreaker. Do you know about different dependency updater before we dive deeper into `udd` and maybe start contributing to it? For now I would not use the current version of `udd` when that means that we have to check for every dependency if version pinning is usable.

I mean the pinning only crashes for dependencies when they're used with a `/` at the end such as `preact` and `preact/` but still, this is one of our main dependencies. 

Let us have a look if we find other (maintained) dependency updaters. Otherwise I would suggest diving deeper into `udd` to see if we can fix that version pinning issue.

**Comment by mbhrznr** [comment]
potentially we could look into [upgrade](https://deno.land/x/upgrade@v1.0.1)?
it has been showcased on discord couple weeks ago.

**Comment by iuioiua** [comment]
Yep! I'm open.

**Comment by iuioiua** [comment]
Candidate: https://github.com/hasundue/deno-molt

**Comment by iuioiua** [comment]
I don't think this is needed anymore, as we now have `deno outdated`.
